src/lib.rs

```diff
@@ -154,16 +173,102 @@ macro_rules! impl_argminmax {
                     }
                     SCALAR::argminmax(self)
                 }
+
+                // As there are no NaNs when NOT using floats -> just use argminmax
+                fn nanargminmax(&self) -> (usize, usize) {
+                    self.argminmax()
+                }
+            }
+        )*
+    };
+}
+
+// Macro for implementing ArgMinMax for floats (f32, f64)
+macro_rules! impl_argminmax_float {
+    ($($t:ty),*) => {
```

Comment:
This was really hairy to parse. Even though it's use internally, let's try to add some documentation here.

I'd also suggest naming the macro parameter `t` to something like `typename`